internal/cache/cachedisk/cachedisk.go

```diff
@@ -0,0 +1,122 @@
+package cachedisk
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/juju/errors"
+	"k8s.io/klog"
+
+	"github.com/bitnami-labs/charts-syncer/internal/utils"
+)
+
+// Cache implements a Cacher using the local filesystem.
+type Cache struct {
+	id  string
+	dir string
+}
+
+// New returns a new Cache object.
+//
+// The internal cache directory path is computed based on the provided
+// workdir. The workdir will contain a hashed subfolder if a unique identifier
+// is also provided.
+func New(workdir string, id string) (*Cache, error) {
+	if workdir == "" {
+		return nil, errors.New("workdir was not provided")
+	}
+
+	hashID := utils.EncodeSha1(id)
+	c := &Cache{id: hashID}
+
+	c.dir = workdir
+	if id != "" {
+		c.dir = path.Join(workdir, hashID)
+	}
+
+	klog.V(4).Infof("Allocating cache dir: %q", c.dir)
+	if err := os.MkdirAll(c.dir, 0755); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return c, nil
+}
+
+// Path returns the path to a cached file named by filename
+func (c *Cache) Path(filename string) string {
+	return path.Join(c.dir, filename)
+}
+
+// Has returns whether the cache contains the file named by filename
+func (c *Cache) Has(filename string) bool {
+	ok, _ := utils.FileExists(c.Path(filename))
+	klog.V(4).Infof("cache hit { op:has, id:%s, filename:%s }", c.id, filename)
+	return ok
+}
+
+// Read reads a cached file named by filename and writes the contents to the
+// provider w writer.
+//
+// If the file does not exist, Read returns an error that satisfies
+// errors.IsNotFound().
+func (c *Cache) Read(w io.Writer, filename string) error {
```

Comment:
I don't see where you are using this method...

Reply:
I am not using this in the code that I have modified. Note that this file was moved from `cache/store.go`